### PR TITLE
fix: modifications for platform compatibility v0.25-dev

### PR DESCRIPTION
**Pull request by shotonoff**
These modifications are needed to be compatible with platform v0.25-dev

### CODE REVIEW

File: dash/src/blockdata/transaction/outpoint.rs [hunk 1]
```diff
@@ -40,6 +40,20 @@ pub struct OutPoint {
 #[cfg(feature = "serde")]
 crate::serde_utils::serde_struct_human_string_impl!(OutPoint, "an OutPoint", txid, vout);
 
+impl From<[u8; 36]> for OutPoint {
+    fn from(buffer: [u8; 36]) -> Self {
+        let mut tx_id: [u8; 32] = buffer[0..32].try_into().unwrap();
```
Comment (fominok) [inline dash/src/blockdata/transaction/outpoint.rs:45]: it's better to use `expect` with a comment inside

Also https://doc.rust-lang.org/std/primitive.slice.html#method.split_at will show your intentions more clearly

File: dash/src/blockdata/transaction/special_transaction/provider_update_registrar.rs [hunk 2]
```diff
@@ -154,10 +155,10 @@ mod tests {
         assert_eq!(provider_mode, expected_provider_update_registrar_payload.provider_mode);
 
         let operator_key_hex = "139b654f0b1c031e1cf2b934c2d895178875cfe7c6a4f6758f02bc66eea7fc292d0040701acbe31f5e14a911cb061a2f";
-        assert_eq!(operator_key_hex, expected_provider_update_registrar_payload.operator_public_key.to_hex());
+        assert_eq!(operator_key_hex, expected_provider_update_registrar_payload.operator_public_key.encode_hex::<String>());
```
Comment (fominok) [inline dash/src/blockdata/transaction/special_transaction/provider_update_registrar.rs:158]:
```suggestion
        assert_eq!(operator_key_hex, hex::encode(expected_provider_update_registrar_payload.operator_public_key));
```

as we need no flexibility on varying types that could be built from char iterators

Same applies to other places in this PR

**Review by fominok — APPROVED** [review]

File: dash/src/blockdata/transaction/outpoint.rs (hunk 1 above)
Comment (QuantumExplorer) [reply dash/src/blockdata/transaction/outpoint.rs:45]: I think it's better to have this be TryFrom instead of From, we shouldn't panic like this.

Comment (QuantumExplorer) [reply dash/src/blockdata/transaction/outpoint.rs:45]: (for the method)

File: dash/src/blockdata/transaction/special_transaction/provider_update_registrar.rs (hunk 2 above)
Comment (QuantumExplorer) [reply dash/src/blockdata/transaction/special_transaction/provider_update_registrar.rs:158]: I don't think what you said works @fominok

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: dash/src/blockdata/transaction/outpoint.rs (hunk 1 above)
Comment (fominok) [reply dash/src/blockdata/transaction/outpoint.rs]: but we won't panic, that's the point, why to carry an error that will never happen?

File: dash/src/blockdata/transaction/special_transaction/provider_update_registrar.rs (hunk 2 above)
Comment (fominok) [reply dash/src/blockdata/transaction/special_transaction/provider_update_registrar.rs]: why do you think so?

Comment (shotonoff) [reply dash/src/blockdata/transaction/special_transaction/provider_update_registrar.rs]: works, I already applied this

File: dash/src/blockdata/transaction/outpoint.rs (hunk 1 above)
Comment (QuantumExplorer) [reply dash/src/blockdata/transaction/outpoint.rs]: Ignore my comment. @fominok is right.

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**